File: crates/sch-tool/tests/fixtures/languages/NormalizeSpaces.java
package texttools;

import tech.schematic.Supertest;
import static tech.schematic.Assumptions.assume;

public final class NormalizeSpaces {
    private NormalizeSpaces() {}

    @Supertest
    public static void normalizingTwiceChangesNothing(String text) {
        assume(text != null);
        String once = Text.collapseSpaces(text);
        String twice = Text.collapseSpaces(once);

        if (!twice.equals(once)) {
            throw new AssertionError("Normalizing twice changed the text.");
        }
    }
}

File: crates/sch-tool/tests/fixtures/languages/NormalizeSpaces.cs
using System;
using Schematic;

namespace TextTools;

public static class NormalizeSpaces
{
    [Supertest]
    public static void NormalizingTwiceChangesNothing(string text)
    {
        var once = Text.CollapseSpaces(text);
        var twice = Text.CollapseSpaces(once);

        if (twice != once)
        {
            throw new InvalidOperationException("Normalizing twice changed the text.");
        }
    }
}
